### PR TITLE
Upgrade steps for flake.nix

### PR DESCRIPTION
**Pull request by amitaibu**

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -188,7 +264,7 @@ Note that the upgrade will drop your existing _local_ database, so make sure to
 
 This means that from now on when adding new packages, you need to do it in a single file - `devenv.nix`
 
-9. **Copy settings from `Config/nix/nixpkgs-config.nix` to `devenv.nix`**
+7. **Copy settings from `Config/nix/nixpkgs-config.nix` to `flake.nix`**
```
Comment (amitaibu) [inline UPGRADE.md:267]: I don't know how to adapt this section

File: UPGRADE.md [hunk 2]
```diff
@@ -132,63 +206,107 @@ Note that the upgrade will drop your existing _local_ database, so make sure to
...
+            }
+        )
+    { src = ./.; }).defaultNix
     ```
 
-This means that from now on when adding new packages, you need to do it in a single file - `devenv.nix`
+This means that from now on when adding new packages, you need to do it in a single file - `flake.nix`
 
-9. **Copy settings from `Config/nix/nixpkgs-config.nix` to `devenv.nix`**
+8. **Copy settings from `Config/nix/nixpkgs-config.nix` to `flake.nix`**
```
Comment (amitaibu) [inline UPGRADE.md:309]: The section needs updating, but I don't know how

Comment (mpscholten) [reply UPGRADE.md:309]: Here's an example using `ihp.dontCheckPackages = [ "my-failing-package" ];`:

```
{
    inputs = {
        ihp.url = "github:digitallyinduced/ihp?ref=abb513016b372f9f76b6c95caed66def536a885a";
        ihp.flake = false;

        nixpkgs.url = "github:NixOS/nixpkgs?rev=a95ed9fe764c3ba2bf2d2fa223012c379cd6b32e";

        systems.url = "github:nix-systems/default";
        devenv.url = "github:cachix/devenv";
    };

    outputs = { self, nixpkgs, devenv, systems, ihp, ... } @ inputs:
        let
            devenvConfig = { pkgs, inputs, config, ... }: {
                # See full reference at https://devenv.sh/reference/options/
                # For IHP specific options, see https://ihp.digitallyinduced.com/Guide/package-management.html

                imports = [ "${inputs.ihp}/NixSupport/devenv.nix" ];
                
                ihp.enable = true;
                ihp.projectPath = ./.;

                ihp.haskellPackages = p: with p; [
                    # Haskell dependencies go here
                    p.ihp
                    cabal-install
                    base
                    wai
                    text
                    hlint
                ];

                packages = with pkgs; [
                    # Native dependencies, e.g. imagemagick
                ];

                ihp.dontCheckPackages = [ "my-failing-package" ];
            };
            
            releaseEnv = pkgs: import "${ihp}/NixSupport/default.nix" {
                ihp = ihp;
                haskellDeps = (devenvConfig inputs).ihp.haskellPackages;
                otherDeps = p: (devenvConfig inputs).packages;
                projectPath = ./.;
                includeDevTools = false;
                optimized = false;
            };
            forEachSystem = nixpkgs.lib.genAttrs (import systems);
        in
            {
                devShells = forEachSystem (system: {
                    default = let pkgs = nixpkgs.legacyPackages.${system}; in devenv.lib.mkShell {
                        inherit inputs pkgs;
                        modules = [devenvConfig];
                    };
                });
                packages = forEachSystem (system: { default = releaseEnv nixpkgs.legacyPackages.${system}; });
            };
}
```

Comment (amitaibu) [reply UPGRADE.md:309]: Where does the `doJailbreakPackages` go, under `ihp.dontCheckPackages`?

Comment (mpscholten) [reply UPGRADE.md:309]: yes below like this:
```nix
ihp.dontCheckPackages = [ "my-failing-package" ];
ihp. doJailbreakPackages = [ "some-package" ];
```

Comment (amitaibu) [reply UPGRADE.md:309]: done, added a shorter diff

**Comment by mpscholten** [comment]
Thanks 👍

**Pull request merged**